### ISSUE TITLE
Suggestion: Triage capacitor and Capital Cap transfer buff

### ISSUE THREAD

**Issue by BlodBladelicker** [kabivel/Sins-of-Empires#200]
For the sake of better triage, I would like to suggest the Cap Regen rate of Triage ships and the Capital Transfer ability gets buffed. 
Currently a triage(or at least the Ninazu) is very capable of capping itself out at which point it becomes near impossible to get its capacitor to a usable state again.
I think this is partially caused by the cap transfer ability not transferring enough capacitor(600 when a single rep cycle takes over 1000 cap) and also the cap transfer ability requiring 300 capacitor of its own. This causes the triage ships to get stuck in a cycle of transferring cap to another ship, capping themselves out again as its the ability with the lowest capacitor cost. 

I tested it in the previous EVE mod, with 50 remote-cap augorors with cap transfers costing capacitor of their own, and even in such a large blob eventually they all get capped out as units on the far end of the blob don't get cap transferred enough, which trickles eventually into the core of the blob causing them all to have too little cap in the end.

**Comment by Joeyxxx** [comment]
This certainly needs looking into, the AI builds many FAX's and they're essentially useless in most situations.

The idea behind this was to create a "cap-chain" but it doesn't work very well on the Sins engine.

I'm behind this suggestion, it's worth testing.

**Comment by smudge202** [comment]
What version have you guys been playing of late? I tend to try out the latest changes so I'm not sure if my observations are valid compared to master, let alone offical releases.

I have almost no problem at all with FAX. My promotion order for them is typically:

1. Rep
2. Triage
3-5: Rep
6-9: Cap
10: Triage (one day there will be a point to this...)

I do push heavily into Mil research, but even my early game FAX's with only the first cap upgrade and no culture boosts don't often struggle unless having to sustain triage mode for 10+ mins. I try to keep them at least in pairs to feed one another for infinite. By late game with all cap and culture research, I don't think it's even possible to cap a pair of FAX out?

The cruisers, sure. They probably need looking at, I haven't had much luck with them. The game engine likes to have all of them throw all their cap at one cruiser, massively wasting cap. That cruiser than sends cap to one of the others, causing them all to transfer cap back to the same cruiser. /sigh Computer isn't too good with cap chains in those situations.

I would like to see better buffs from Triage modules though (Cooldown **+** Amount bonuses). Currently a titan, not using super weapons (guns only), can go through a Dread with 4 FAX's in Triage mode trying to keep it up and full resists. I'm not sure that would be true in Eve?

**Comment by BlodBladelicker** [comment]
I'll go do some testing with rep power and cap transfer ability, using Caldari as reference if you pair them, and what happens if you add more to the mix. More exact testing details after it's done.

**Comment by Joeyxxx** [comment]
Perhaps the reps do need to be stronger, as the ship AI which uses abilities tends to repair the wrong ships. I've seen many dieing ships simply ignored by my FAX's as they repair each other.

**Comment by smudge202** [comment]
I'm not sure what priority system the AI uses for reps. I have a feeling it's done by ship size? A damaged titan will often sit getting repped despite not being attacked whilst other ships are mercilessly picked off.

**Comment by BlodBladelicker** [comment]
I think it uses a system of importance by ship size as you said, though I am not sure how it works. And also a system of closest to farthest.
But I am not sure if that really is the case either.

**Comment by BlodBladelicker** [comment]
Testing done! These are the notes I took while testing. I'm sure someone else could write much better notes but these are mine. 
Opinion down below of this scenario

Scenario 1:
Mine:
Phoenix + 2 Minokawa, Lvl 4
Phoenix: 	        siege: 	1
			anti:	        1
			Invul:	2
Minokawa:	Triage:	1
			Remote:	2
			Transf:	1

Hostile:
Ragnarok lvl 1 capacitor+tank
2489 DPS
10 lif

Result:
Phoenix dies slowly - ragnarok barely holding
1 minokawa cap stable
1 minokawa slowly bleeding cap

Furthering that same scenario. 
Add 2 minokawa same abilities:
a minokawa can't rep target behind him? moving slightly further away from phoenix, now he can?
1 minokawa bleeding cap a lot quicker than before

With all 4 minos focused on repping him, he stays barely alive

Up all their levels to 10.
phoenix level 10
4/2/4 instead of 1/1/2 before
easily tanks

minos level 10:
2/4/4 instead of 1/2/1 before
still bleeding cap on the same minokawa(not receiving any remote cap?)

Observed resists on phoenix: 49% off invul, 84% with invul. Phoenix tanking easily now during 84% moments.

Hostile's 10 lifs: one if em bleeding cap(most outer edge lif) but not at an alarming rate, down to ~50% of cap after 5-10 mins I'd guess

>  I'd say the minokawa's were doing well cap-wise for the most part, except for 1 minokawa that was bleeding cap at a non-threatening rate(however, this was without needing to use the self-rep as well. That would have depleted the cap quickly). The rep power was rather low however as Smudge said, needing 4 minos to keep 1 invuln'd phoenix alive. 
The minokawa not being able to rep behind him was a bit strange. He refused to do anything else as the AI was waiting for him to start his rep cycle, which he couldn't. When he moved 5 meters further away from the phoenix he could rep behind him.

**Comment by BlodBladelicker** [comment]
Opinion down below of this scenario

Scenario2:
subcap battle
Gallente vs minmatar

Gallente:
thanatos4 	x2 0/1/2/1(self-rep/smartbomb/damage-bonus/mindlink)
ninazu4		x2 1/2/1(triage/rep/transfer)
dominix 	        x20
brutix		x5
vexor		x30
execuror	       x10

minmatar:
nidhoggur4 	x2	(unknown)
lif4		        x2    (unknown)
tempest		x20
hurricane	        x6
stabber		x30
scythe		x10

execurors only using 1 rep due to beam length time? They seemed to wait till beam time of a rep was over before using next rep. But since beam was as long as cooldown of that one ability they never used the other rep.
still issues with units being behind triage - cant rep - ai does nothing until that target is repped somehow
2 thanatoses worth of firbolg break through dual triaged lif reps, nidhoggurs die. Lif cap is not having any issue
non-bugged ninazu stuck at 0-1000 cap, bugged still at 44k. 
moved problem-causing-unit for bugged ninazu, now repping properly. Down to 0-1000 cap as well 
Subcaps dying left and right for both sides. Lifs too focused on nidhoggurs to rep subcaps, ninazus focus on subcaps as caps aren't taking damage. Domis start tanking as minmatar's subcaps die


> Again rep power problems when repping capitals. Simply not capable of holding up even with just 6 fighters on the target. 
Also again the rep problem on the triage. Rather frustrating, not sure why it happens. 
A possible help for the issue would be having auto attack on logi and triage. With triage reps having a range of 20000+ and an attack range of 20000 for example. Meaning they will sit and triage at 20000 from hostiles and have friendlies always in front of them thus avoiding the issue. Logi staying at edge of planet grid is also an issue as they can't rep anything from 50000 away, giving them an attack range of 12500-15000 and a rep range of 15000+ would fix this too.
Ninazus were going down in cap at a ridiculous rate for some reason. Despite having I believe similar stats to Minokawas/Lifs. The 100 cap/s from triage and cap transfers kept em capable of repping with some degree of certainty, but if there were more ninazus with some falling out of cap transfer chain those would be useless to the fleet. If they were to not have enough cap for triage at any point they'd become completely useless as their own regen would be too weak to get back into triage.

**Comment by smudge202** [comment]
Nice test results, thanks for taking the time to both conduct and write it up! :+1:

It sounds like your observations more or less matched my own? I haven't played Eve in so long that I'm hesitant to categorically say the rep amounts are under-powered, but I'm fairly sure that is the case. The problem with powering up those reps however:

In eve, fleets don't often stick to a single race of ship (yes, a bulk of dps will all be the same ship typically, but logistics, ewar, and other key elements will use the race best tailored to the fleet at hand). To break logistics there are 4 mechanisms:

1. Out-DPS the reps
2. Cap Warfare - only available to Amarr in Sins
3. ECM - only available to Caldari in Sins
4. Sensor Dampening (and forcing range) - Sensor Dampening only effects weapons in Sins (not abilities afaik)

The point being, as much as I'd like to see more reps, we need to consider balance, especially for Gal/Min who have no real rep-breaking abilities beyond pure DPS.

Your second scenario write up, I *thought* that FAXs tend to sit on edge of gravity well not because of weapon ranges but because they activate triage as soon as an enemy is present, which cuts their engines? I might be wrong, but before reading your summary, that was my understanding. I typically sat my Capitals at the back of fleet fights but always moved them towards the enemy fleet, manually moving sub-caps infront so I hadn't noticed the backwards-repair issue you've found.

>"rep power problems"

^ did you mean rep amounts being insufficient or was the capacitor running dry?

**Comment by BlodBladelicker** [comment]
Faxes enter triage upon combat. But if they never activate their weapons or are never getting hit, they are never in combat according to the sins engine. And due to not having auto attack enabled, they just sit at the edge of grid.

The rep power was insufficient, the dry capacitor didn't help but wasn't the cause of the rep power problems.

**Comment by Joeyxxx** [comment]
The 'beam length time' is interesting. Could anybody shed some light on this and what it does?

**Comment by BlodBladelicker** [comment]
I haven't flown triage myself, but I know that triage is an important healer even in a 100 vs 100 battle. 

[https://www.youtube.com/watch?v=i5r3582fTX0](url)
Shows how faxes can even stop a dreadbomb entirely these days due to their rep power especially in large number. Requiring all DPS to be put on the triage.
I do agree that the sins engine/AI makes it harder to balance. But given the alpha/dps possible in this mod, I don't think it's too big of a deal if the rep gets buffed to 3k or more.
Testing it myself:
Hostiles have naglfars
I have apostles
Maxed research
lvl 4

Naglfars:
Siege 1
Anti-capital 1
Invul 2

Apostles:
Triage 1
Remote 2
Transfer 1

Current:
3 apostles vs 1 naglfar: apostles win
3 apostles vs 3 naglfar: naglfars win

3k rep power:
3 apostles vs 3 naglfar: apostles win
3 apostles vs 5 naglfar: naglfar win
6 apostles vs 5 naglfar: apostles win
6 apostles vs 5 naglfar+50 canes+50 tornado: winmatar
6 apostles vs 5 naglfar+50 typhoons: winmatar
13 apostles vs 5 naglfar+hel+3 ragnarok(doomsdays not auto casting nor auto cast capable?)+3 avatar+50 typhoons: Slow at the start, but apostle destruction sped up a lot as more died, but certain victory for winmatar (went a bit overboard perhaps)

I'd argue that needing more dreads than hostile triage isn't unfair, given that there are also many subcaps on field to provide DPS.

**Comment by BlodBladelicker** [comment]
> The 'beam length time' is interesting. Could anybody shed some light on this and what it does?

For the player, it is how long we are able to see the repair beam from 1 ship to another.
I needed to delve into the files, but it is the 'Finish Condition' "TimeElapsed" of BuffRemoteArmorRepairerLogiCruiserSelf. Currently it is at 5 (seconds), with an ability cooldown of 6. This MIGHT be causing the AI(who's tick is possibly horrendous) to only use 1 rep. Making the condition down to 2.5 allows the AI to use both reps - when it feels like it.
Because the AI is annoying and sometimes refuses to rep a target at all even if it is the ONLY target on field that needs reps. Or still only uses 1 rep. But at least it has like a 10-40% chance to actually use the second rep then. God damn this AI
Putting it further down to 1.0 matters none.

**Comment by Joeyxxx** [comment]
Would you say that decreasing it to 2.5 is an improvement?

**Comment by smudge202** [comment]
For reference:

Apostle in Eve will typically have 3x Capital Remote Rep (let's assume vanilla tech 1 rep). 
Without Triage mode active but with Level 5 skills, that would be about 900HP x 3 modules every 12 seconds (225 armour/sec).
With Triage mode active and level 5 skills, it's 5000HP x 3 modules every 3 seconds (5000 armour/sec).

In Sins, there are 2 abilities that only decrease in antimatter cost as levelled so level is not relevant to the above. The Triage mode in Sins also increases your passive antimatter restore rate which, so again, not really a relevant comparison there.
Without Triage mode active, it's 7920HP x2 abilities every 22.5 seconds (704 armour/sec)
With Triage mode active, it's 7920HP x2 abilities every 6.4/5.8/5.6/5.4 seconds (2475-2933 armour/sec).

As you can see, Triage mode in Eve is **massively** more important than it is in Sins. And that's ignoring the fact that Sins ships don't model resistances so have armour values reflecting Eve's *effective HP*, so the Eve numbers are even higher than those shown.

**Comment by BlodBladelicker** [comment]
You're mistaken Smudge202, it's not 7920 x 1, it's 1094 x 1 every 9 seconds outside of triage. The 7920 is the self-heal ability of triage(and others)

**Comment by smudge202** [comment]
Oh crap, good point, mis-clicked the file and then didn't read it:

https://github.com/kabivel/Sins-of-Empires/blob/master/GameInfo/BuffRemoteArmorRepairerCapitalTarget.entity#L27-L30

I'll update the post above with correct numbers

**Comment by BlodBladelicker** [comment]
Since this has become a bit more of a triage discussion in general. 
I think that the way one of the previous EVE mods handled triage in a more enjoyable way. For example the Archon: 

Total hull of ~50k at max level(so stats are obviously lower than they should be for the current mod as DPS was also lower back then)
Triage + Remote Rep + Cap Transfer + Drone Dock ability
Triage had 2 levels, costed capacitor, and restored hull and shield points(100/150) and reduced cooldowns by 1000%. Cooldown of 520 seconds and a duration of 45. Giving it some time to move on the battlefield between cycles.
Remote Rep had 4 levels, didn't cost any cap(750), had a cooldown of 25 seconds outside of triage, making it virtually useless outside of triage, and only repaired 1000-1500.
Cap Transfer had 4 levels, didn't cost any cap, had a cooldown of 45 seconds outside of triage, and restored 300-650 cap. 
This meant that you could neut out the hostile triage preventing it from going back into triage, effectively neutering it, which is impossible in the current setting. With the high neut cooldown and the 100cap/s from triage not even 20 bhaalgorns could keep a single triage capped out unless you were to micro the ability timings. 

Now what am I actually getting at? I think that removing the self-rep is a good thing. At it could be easily added to the triage module buff instead. Opening up another ability slot or making level progression easier and better for AI players(though removing choice for players, if one cares about that)
I also think that removing the high capacitor regen is a good thing so that useful neut warfare can enter the mod. Making triage the only antimatter costing ability, with a high enough antimatter cost, could result in the ability to neuter hostile triage. 
Neuts can easily be added to battleships, tempest, dominix, armageddon and raven for example so each race has neut power late game.

**Comment by Joeyxxx** [comment]
Neuts could even be added to the new Pirate ships - making them very late game indeed.

I'm on board with your idea of reverting back to the previous triage mechanics - what we have at the moment simply doesn't work.

If Smudge agrees, I think we should make a definitive list of changes for Kab so that we can test it on a non-master branch.

**Comment by smudge202** [comment]
I'm working on the following proposal, it's not ready yet and will be edited profusely. It works with the current mechanics but I'm going through numbers constantly updating them as I refine the math.

## Proposal

I want to have a crack at re-balancing the FAXs. I'll do this for no more than 1 week on a fork of this repo so that anyone that has the time can help me test it? After which I'll either kill of the branch (failed) or PR it back here for @kabivel to review (hopefully alongside some positive feedback from people that tested it)

I'll list here the abilities I'm going to change relevant to Force Auxilliaries (I'm not going to touch Logi cruisers... for now).

### Triage Mode

Available at levels: 1, 10
Autocast: No
Duration: 120 seconds
Effects:
* Disables Phase Jump
* Disables Movement
* **Cannot be Remote Assisted**
* Cannot be ECM'd
* +100/120% (Local) Capital Repair Amount
* +400/450% Remote Capital Repair Amount
* +400/450% Remote Capacitor Transfer Amount
* -50% Ability Cooldown (i.e. half duration)

> _**Notes**: Between this and the buffs below for Remote Repair/Cap Transfers, the FAX should be able to push out a truly astonishing amount of HP. However, by introducing the "Cannot be Remote Assisted" it gives all races a viable tactic to combat FAXs - kill the triaged FAXs. Sure, they'll be able to run local repairs better than most, but without cap bonuses and cap transfers, nor remote armour, they won't hold up long on themselves. I believe this reflects current Eve bonuses pretty accurately?

### Capital Remote Repair

Available at levels: 1, 2, 3, 4
Autocast: Yes
Duration: 12 seconds
  * With Triage: 6 seconds
Effects:
* Repairs target for 2250/2850/3300/3600HP 
  * With Triage L1: 9000/11400/13200/14400HP
  * With Triage L2: 10125/12825/14850/**16200**HP

### Capital Repair

Available at levels: 1, 2, 3, 4
Autocast: Yes
Duration: 30/27/24/22 seconds
  * With Triage: 15/13/12/11 seconds
Effects
* Repair self for: 9500/11500/13000/13500
  * With Triage L1: 19000/23000/26000/27000HP
  * With Triage L2: 20900/25300/28600/**29700**HP

### Capital Remote Capacitor Transmitter

>Intentionally not going to list the specifics of this ability because I haven't looked at antimatter costs at all yet. However, my goals would be:
>
>* With full culture and capacitor research, inside culture (so MAX bonuses) running everything whilst in triage would suck a FAX dry after ~30-45secs (yes, that's right; it simply won't be feasible to run everything whilst in triage)
>* With full research, inside culture running triage + **remote** rep only, would deplete a FAX after ~120secs (i.e. after 1 triage cycle, if using reps flat out, it'll be dry and need to come out of triage to recover)
>* With full research, inside culture running triage + **local** rep only, would not deplete the FAX (i.e. passive antimatter restore would **just** keep it afloat, making it very easy to neut the FAX causing it to topple into being unable to rep)
>* Without research, outside culture, etc, scale back the times by the percentages currently in use for antimatter/capacitor bonuses

## Todo:

* All HP amounts are derived from Eve without weighting for effective HP comparisons between Sin<->Eve yet. Will need updating
* Per feedback

**Comment by Joeyxxx** [comment]
I've noticed that ships with self-repair abilities often begin repairing when it's too late, constantly repair or fail to repair themselves even with autocast on.

This is not part of your balancing and mechanic changes but it's something to bare-in-mind during this exercise.

**Comment by smudge202** [comment]
Agreed, the ticks that the AI work to are indeed very slow. I don't think it's possible for them to use abilities "on-cooldown" because the AI ticks simply don't coincide. I'm hoping to OP the abilities a little to account for this, but not so much that players micro-managing can truly take the mick

**Comment by BlodBladelicker** [comment]
Very nice write up Smudge202! A few points I can already see be a major problem however. For my gameplay style at least. 

> No Autocast

You need autocast to be enabled to be able to play against AI without it being a roflstomp unless they bring 10.000 BSs against your 100. I am someone who plays pretty much solely against AI so I am a big advocate for making the AI challenging or at least fun to fight.
As reference, I got several me vs AI battle videos on youtube, and I have about 500 hours of gameplay into the previous EVE mod 99.9% against AI. I am very serious about wanting a good AI. 
A benefit of having a good AI is meaning less micro management needed for the player.
In a late game battle, do you want to micro-manage your 60 triages+dreads+supers+carriers? I sure don't. I got a whole galaxy and subcap fleet to manage.

The same issue is present in the current mod. Too many abilities are not autocast enabled, causing the AI to not use it ever(RIP Ragnarok's Doomsday for example)

> Triage capping itself out without outside pressure

This doesn't happen in EVE, why should it in this mod? A triage that is properly managing its cap won't face many issues unless it is also facing neut pressure or needs to use both self rep and all 3 remote reps for a prolonged time. This also only increases the terribleness of macromanaging instead of micromanaging.

Sorry if it sounded a bit unfriendly, definitely not the intention.

**Comment by BlodBladelicker** [comment]
> Agreed, the ticks that the AI work to are indeed very slow. I don't think it's possible for them to use abilities "on-cooldown" because the AI ticks simply don't coincide. I'm hoping to OP the abilities a little to account for this, but not so much that players micro-managing can truly take the mick

#211 I think this issue may very well be related to that. I'm looking at my past videos now of the previous EVE mod to see how the AI seemed to behave there in terms of ability speed. If I can't see it clearly enough I'll simply play another round on that version and watch it closely.

**Comment by smudge202** [comment]
Not at all, perfectly valid points. I will endeavour to balance this all so that Autocast can in fact be enabled. I had already had some thoughts on how to make that possible (contrary to Eve, make activation of Triage the really expensive bit, which means the AI wouldn't try to activate Triage - because not enough cap to do so - unless it will have enough to run abilities per the above goals)

**Comment by kabivel** [comment]
what a hell
dont have time to read all this
someone make a resume and give numbers

**Comment by Joeyxxx** [comment]
@kabivel the guys are still trying to figure this one out, there are many considerations.

Work-in-progress.

**Comment by smudge202** [comment]
I've been through 3 hefty revisions of this so far locally. The problem is balancing on a slightly broader scale (looking at DD balance, Bombers, Dreads, etc... it's proving tricky so yes, WIP)

**Comment by Joeyxxx** [comment]
Any news on this one?

**Comment by smudge202** [comment]
Started a new job on Monday this week so spare time has been at a premium. Hoping to have a couple full games this weekend to look into my latest round of balancing.

**Comment by Joeyxxx** [comment]
@smudge202 have you had a chance to take a look?
